src/main/java/com/example/WorldOfAnimals/controllers/AnimalBreedController.java

```diff
@@ -2,49 +2,141 @@
 
 import com.example.WorldOfAnimals.dto.AnimalBreedDTO;
 import com.example.WorldOfAnimals.dto.AnimalBreedRequestDTO;
+import com.example.WorldOfAnimals.dto.AnimalBreedRequestPutDTO;
+import com.example.WorldOfAnimals.dto.ErrorResponseDTO;
+import com.example.WorldOfAnimals.exceptions.AnimalBreedNameDuplicateException;
 import com.example.WorldOfAnimals.services.AnimalBreedService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
 import lombok.AllArgsConstructor;
-import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
 import java.util.List;
 
+@Tag(
+        name = "Animal breeds",
+        description = "Manage animal breeds by define " +
+                "(existed in base) animal type"
+)
 @RestController
 @RequestMapping("/animals/breeds")
 @AllArgsConstructor
 public class AnimalBreedController {
 
     private AnimalBreedService service;
 
+    @Operation(
+            summary = "Create new breed of animal",
+            description = "For successfully creating animal breeds," +
+                    " the animal type (ID) must be existed in base"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Breed created"),
+            @ApiResponse(responseCode = "422", description = "Duplicate unique animal breed name",
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = ErrorResponseDTO.class))}),
+    })
     @PostMapping
-    @ResponseStatus(HttpStatus.CREATED)
-    public void createBreed(@RequestBody AnimalBreedRequestDTO breedRequestDTO) {
-        service.save(breedRequestDTO);
+    public ResponseEntity createBreed(@RequestBody AnimalBreedRequestDTO breedRequestDTO) {
+        try {
+            service.save(breedRequestDTO);
+            return ResponseEntity.status(201).build();
+        } catch (AnimalBreedNameDuplicateException e) {
```

Comment:
It one posible how to handler exception but maybe need use ControllerAdvice ?

Reply:
Done